### ISSUE TITLE
Have you tried testing your model on VGGFace2? My test shows incredibly bad result. Need help in verification.

### ISSUE THREAD

**Issue by yxchng** [wy1iu/sphereface#98]
This issue is not meant to degrade your work. I may have done something wrong in my test and I need some help in verifying if my test procedure is right. 

The dataset is: http://www.robots.ox.ac.uk/~vgg/data/vgg_face2/. I used your face_detect_demo.m and face_align_demo.m to detect and align faces in vggface2_train (8631 people) and vggface2_test (500 people). 

Following megaface procedure with slight modification, I make vggface2_train the test set and vggface2_test the distractor. 

For each identity in test set, I compute the max intraclass distance of one photo (reference) with the rest of the photos. Then, I compute the min interclass distance of this reference with the rest of the photos in both the test set and the distractor, excluding photos of the identity being tested.

A good face recognition algorithm should have min interclass distance > max intraclass distance. Each image that has this property is considered correct.

Let the number of correct photo be C. Then, if there are N total photos in the dataset, the accuracy will be C/N.

The code for my test is here: https://gist.github.com/yxchng/dec1ec6fe306082684af70d85f2590e2.
I save each features into a text file and read them in the code. You may do it your way if you want to test it. Hopefully, you can check that there is no logic error in my code.

**The result I get for sphereface is an incredible 0%. Do you think this result is expected given my test setting? What am I doing wrong here.**

**Comment by wy1iu** [comment]
SphereFace can work well with VGGface2. Previously, we have successfully trained SphereFace on VGG2. Besides that, the results of [InsightFace](https://github.com/deepinsight/insightface) also has validated this. Your testing procedure may somehow be buggy.

**Comment by yxchng** [comment]
@wy1iu Training on VGG2 and testing on it is different from training on CASIA-WebFace and testing on VGG2. And InsightFace has not validated this. 

Training on VGGFace2 and testing on it is a closed-set problem. I am verifying open-set problem.

And as I said, I use your preprocessing code and model.

Do you need me to write in Chinese? I can write the issue in Chinese because you do not seem to have read and understand my post.

**Comment by yxchng** [comment]
这个不是一个传统的测试方式。没有人这么测试过，也许因为结果会是百分之零。 

测试代码绝对没有问题: https://gist.github.com/yxchng/dec1ec6fe306082684af70d85f2590e2.

这个测试方式要每张图类内最远的距离近于类间最近的距离才算对。vggface2平均每张图类内300张图左右，类间3百万张

这个测试说明没有一张图符合这个条件:

`max_interclass_margin < min_interclass_margin`

我所有的preprocessing都是用你的代码，就算有差也是微差: https://github.com/wy1iu/sphereface/issues/93, 
不会影响结论.

为什么要这么测试? 因为要高机密的应用，不允许任何误识别。这个测试是最严谨的。

**Comment by wy1iu** [comment]
如果我没有理解错，你是用casia训练的模型，然后直接用vgg2的数据去测试，而protocol就是用的你描述的这个。我没检查你的测试代码（假设你的预处理、测试等部分没问题），但是我觉得你还是可以做个sanity check，你可以拿除了sphereface之类的模型，比如center loss，比如普通的softmax loss，做一个同样的测试，看看结果怎么样。我觉得在casia上训练的模型（比如我们提供的sphereface pretrained model），在vgg2上不能达到你说的那种“类内最大距离小于类间最小距离”，这个也说不定是有可能的。
